src/devices/Ads1115/Ads1115.cs

```diff
@@ -104,14 +220,255 @@ private void SetConfig()
+                // We need to wait two cycles when changing the configuration in continuous mode,
+                // otherwise we may be getting a value from the wrong input
+                _i2cDevice.Write(writeBuff);
+                WaitWhileBusy();
+                configHi &= 0xFE; // Clear last bit
+                writeBuff[1] = configHi;
+                _i2cDevice.Write(writeBuff); // And enable continuous mode
+            }
+        }
+
+        /// <summary>
+        /// Resets the comparator registers to default values (effectively disabling the comparator) and disables the
+        /// Alert / Ready pin (if configured)
+        /// </summary>
+        private void DisableAlrtReadyPin()
+        {
+            _comparatorQueue = ComparatorQueue.Disable;
+            SetConfig();
+            // Reset to defaults
+            Span<byte> writeBuff = stackalloc byte[3]
+            {
+                (byte)Register.ADC_CONFIG_REG_LO_THRESH, 0x80, 0
+            };
+            _i2cDevice.Write(writeBuff);
+            writeBuff = stackalloc byte[3]
+            {
+                (byte)Register.ADC_CONFIG_REG_HI_THRESH, 0x7F, 0xFF
+            };
+            _i2cDevice.Write(writeBuff);
+            if (_gpioController != null)
+            {
+                _gpioController.UnregisterCallbackForPinValueChangedEvent(_alrtRdyPin, ConversionReadyCallback);
+                _gpioController.ClosePin(_alrtRdyPin);
+                _gpioController = null;
+                _alrtRdyPin = -1;
+            }
+        }
+
+        /// <summary>
+        /// Write the two comparator registers
+        /// </summary>
+        /// <param name="loThreshold">High threshold value (unsigned short)</param>
+        /// <param name="hiThreshold">Low threshold value (unsigned short)</param>
+        private void WriteComparatorRegisters(short loThreshold, short hiThreshold)
+        {
+            Span<byte> writeBuff = stackalloc byte[3]
+            {
+                (byte)Register.ADC_CONFIG_REG_LO_THRESH, (byte)(loThreshold >> 8), (byte)(loThreshold & 0xFF)
+            };
+            _i2cDevice.Write(writeBuff);
+            writeBuff = stackalloc byte[3]
+            {
+                (byte)Register.ADC_CONFIG_REG_HI_THRESH, (byte)(hiThreshold >> 8), (byte)(hiThreshold & 0xFF)
+            };
+            _i2cDevice.Write(writeBuff);
         }
 
         /// <summary>
-        /// Read Raw Data
+        /// Enable conversion ready event.
+        /// The <see cref="AlertReadyAsserted"/> event fires each time a new value is available after this method is called.
+        /// </summary>
+        /// <param name="gpioController">Reference to the GPIO controller</param>
+        /// <param name="pin">Pin to which the ALRT/RDY Pin is connected</param>
+        /// <exception cref="InvalidOperationException">The conversion ready event is already set up</exception>
+        public void EnableConversionReady(GpioController gpioController, int pin)
```

Comment:
how often would the user call this with different gpioControllers? would it make sense to take this in the constructor with default value and change this to paramterless?

Reply:
Of course, you would always use the same controller (since apparently, you have to wire a pin for that). I guess it would be more consistent with other bindings to make it an optional parameter in the constructor.